### PR TITLE
Don't panic when get refresh rate failed

### PR DESCRIPTION
**Pull request by xiaopengli89**
- [x] Tested on all platforms changed

### CODE REVIEW

**Comment by kchibisov** [comment]
Was it failing with external display or something?

**Review by kchibisov — CHANGES_REQUESTED** [review]
Is it happening when the monitor was removed by handle was something kept? It seems strange that these calls fail.

**Comment by kchibisov** [comment]
We should avoid panic here if it could appear in the world ofc, it just assumed that those calls can't fail, but I guess the handle became invalid somehow?

You should also add a `CHANGELOG` entry for that change.

File: src/platform_impl/macos/monitor.rs
Comment (xiaopengli89) [reply src/platform_impl/macos/monitor.rs:229]: Failed to get display link when the display is removed.

**Comment by xiaopengli89** [comment]
The error message:
```
assertion failed: `(left == right)`\n left: `-6670`,\n right: `0`
...
winit::platform_impl::platfrom::monitor::MonitorHandle::refresh_rate_millihertz
```

```
InvalidDisplay                    -6670              The specified display is invalid
```

**Review by madsmtm — CHANGES_REQUESTED** [review]
Thanks, could I get you to describe in the code each case where we now return `None` why it can fail? That helps a lot when making further changes in the future!

**Comment by xiaopengli89** [comment]
I add some comments to the code.

File: src/monitor.rs
Comment (kchibisov) [inline src/monitor.rs:147]: Could you wrap akin other doc comments here?

Comment (xiaopengli89) [reply src/monitor.rs:147]: What doc comments should I put here?

Comment (kchibisov) [reply src/monitor.rs:147]: I'm not saying to put, I'm saying that you're line is too long, make it 80 chars wide like other doc comments.

Comment (xiaopengli89) [reply src/monitor.rs:147]: Got it.

File: CHANGELOG.md
Comment (kchibisov) [inline CHANGELOG.md:31]:
```suggestion
- On macOS and Windows, fixed crash when couldn't get monitor refresh rate.
```

**Review by kchibisov — CHANGES_REQUESTED** [review]
Would need this change and rebase.

**Review by kchibisov — APPROVED** [review]

**Comment by xiaopengli89** [comment]
Does this pr need any other changes? @madsmtm

**Comment by kchibisov** [comment]
If your patch isn't getting merged there might be a maintenance issue related to that or it might be on a `queue`. That's at least what I can say for winit. It'll be merged before the next release for sure. Just not right now, so please don't try to get extra attention to it.

Right now we want to fix our CI, and then I'll merge it with other patches together.

**Comment by xiaopengli89** [comment]
I'm sorry to disturb the maintainer, our customers have reported this problem (this problem is very important to our customers), I hope we can solve this problem as soon as possible, we are currently maintaining a fork version of winit to solve this problem; but I still hope to use the official version of winit (higher code quality, more complete functions).

> It'll be merged before the next release for sure.

I feel at ease with this sentence, I will wait for it .
Finally thanks to the maintainers of winit for their work.

**Review by madsmtm — APPROVED** [review]
I think it's fine as-is (though yeah, waiting on CI stuff), thanks for the contribution and your patience!

**Pull request merged**
